vaadin-grid-pro-flow-parent/vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/EditColumnConfigurator.java

```diff
@@ -150,21 +152,37 @@ public <V> Column<T> custom(HasValueAndElement<?, V> component,
             attachRegistration.remove();
             attachRegistration = null;
         }
+        // Need to call on attach to make sure that the edit mode renderer is
+        // set in case the GridPro is detached and attached again
         attachRegistration = column.getElement()
                 .addAttachListener(e -> setEditModeRenderer(component));
 
-        column.getElement().getNode()
-                .runWhenAttached(ui -> ui.beforeClientResponse(column,
-                        context -> setEditModeRenderer(component)));
+        // Calling setEditModeRenderer here in case the GridPro is already
+        // attached and the column is added later
+        // This is needed because in this case the attach listener is not called
+        setEditModeRenderer(component);
+
         return configureColumn(valueProvider, (item, ignore) -> itemUpdater
                 .accept(item, component.getValue()), EditorType.CUSTOM,
                 component);
     }
 
     private <V> void setEditModeRenderer(HasValueAndElement<?, V> component) {
-        UI.getCurrent().getPage().executeJs(
-                "window.Vaadin.Flow.gridProConnector.setEditModeRenderer($0, $1)",
-                column.getElement(), component.getElement());
+        if (editModeRendererRequested) {
+            return;
+        }
+        editModeRendererRequested = true;
+        column.getElement().getNode().runWhenAttached(ui -> {
+            ui.beforeClientResponse(column, context -> {
+                if (!editModeRendererRequested) {
+                    return;
+                }
```

Comment:
**nit:** Is this check necessary? Because `runWhenAttached` is the only place where `editModeRendererRequested` can be set to `false`.

Reply:
This is the same case in the `scheduleHeaderRendering`. My understanding is to safeguard any unlike case of two callbacks scheduled even with the early exit at the beginning of the method.